src/audio/module_adapter/module_adapter.c

```diff
@@ -1353,6 +1445,12 @@ int module_adapter_unbind(struct comp_dev *dev, void *data)
 	if (source_index >= 0)
 		mod_source_info->sources[source_index] = NULL;
 
+	/* Re-enable 1 source 1 sink simple copy mode if single source buffer */
+	if (mod->simple_copy && !module_adapter_multi_sink_source_check(dev)) {
```

Comment:
do we need this simple_copy check here? if all we care about is whether a module has multi sources/sinks, it shouldnt matter right?

Reply:
I removed this but, I wonder if it now causes an issue. If a module adapter client has not set simple copy the processing should never be handled by simple copy function. To speed up code, the module_adapter_copy() when calling module_adapter_simple_copy_1to1() is not checking mod->simple_copy.

I think I need to add to module_adapter_multi_sink_source_check() test for mod->simple_copy, and return true if simple_copy is false.